src/test/java/uk/ac/ebi/atlas/experimentpage/tsneplot/TSnePlotServiceTest.java

```diff
@@ -92,9 +93,18 @@ void testFetchPlotWithMetadata() {
         var metadataCategory = "characteristic_inferred_cell_type";
         var metadataValues = ImmutableList.of("neuron", "stem cell", "B cell");
 
-        var randomPointDtos = RandomDataTestUtils.generateRandomTSnePointDtos(NUMBER_OF_CELLS);
-        when(tSnePlotDaoMock.fetchTSnePlotForPerplexity(experimentAccession, perplexity))
-                .thenReturn(ImmutableList.copyOf(randomPointDtos));
+        var randomPointDtos =
+                RandomDataTestUtils.generateRandomTSnePointDtos(NUMBER_OF_CELLS)
+                    .stream()
+                    .map(pointDto ->
+                            TSnePoint.Dto.create(
+                                    MathUtils.round(pointDto.x(), 2),
+                                    MathUtils.round(pointDto.y(), 2),
+                                    pointDto.name()))
+                    .collect(toImmutableSet());
+
+        when(tSnePlotDaoMock.fetchTSnePlotForPerplexity(eq(experimentAccession), eq(perplexity)))
```

Comment:
why do we need to use Mockito argument matcher `eq` here? Shouldn't we keep it simple and just pass `experimentAccession` and `perplexity`?

Reply:
Good catch. Maybe we were playing with arguments and left `eq` [because](https://site.mockito.org/javadoc/current/org/mockito/Mockito.html#3):
```java
verify(mock).someMethod(anyInt(), anyString(), eq("third argument"));
//above is correct - eq() is also an argument matcher

verify(mock).someMethod(anyInt(), anyString(), "third argument");
//above is incorrect - exception will be thrown because third argument is given without an argument matcher.
```

So if at any point we were doing a fuzzy match we set `eq` in the other argument and so on. That’s the only reason I can think of. In any case, simple is good. :smile: